### PR TITLE
Fixed years error and teachers timeoff

### PR DESCRIPTION
**Pull request by karimosman**

### CODE REVIEW

**Comment by karimosman** [comment]
Ready to merge

**Comment by karimosman** [comment]
@svenvanhal

File: Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs [hunk 1]
```diff
@@ -1,67 +1,73 @@
...
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Xml.Linq;
+using Timetabling.DB;
+
+namespace Timetabling.Objects.Constraints.TimeConstraints.Tests
+{
+    [TestFixture()]
+    public class ConstraintTeacherSetNotAvailableTimesTest
```
Comment (svenvanhal) [inline Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs]: Why the change from internal to public here? Class should not be visible outside the testing library.

File: Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs [hunk 2]
```diff
@@ -1,67 +1,73 @@
...
+            mockDB.Setup(item => item.Tt_TimeOff).Returns(mockSet.Object);
+            mockDB.Setup(item => item.HR_MasterData_Employees).Returns(mockSetEmp.Object);
+
+            test = mockDB;
+
+        }
+
+        [Test()]
+        public void TestConstruct()
+
```
Comment (svenvanhal) [inline Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs]: Unnecessary newline

File: Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs [hunk 3]
```diff
@@ -1,67 +1,73 @@
...
+            XElement[] result = constraint.Create(test.Object);
 
-}
+           // Assert.AreEqual(1, result.Count(item => item.ToString().Equals(constraintTest.ToXelement().ToString())));
+            Assert.AreEqual(0, result.Count(item => item.ToString().Equals(constraintTest2.ToXelement().ToString())));
+
+        }
+    }
+
+
```
Comment (svenvanhal) [inline Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs]: Unnecessary newline

File: Timetabling/Objects/YearsList.cs [hunk 4]
```diff
@@ -50,9 +51,14 @@ public override void Create()
             //Creates the different subgroups in eacht group
             foreach (var item in groups)
             {
-                list.Elements("Year").Elements("Group").First(g => g.Element("Name").Value.Equals(item.ClassName)).
-                    Add(new XElement("Subgroup",
-                                     new XElement("Name", item.groupName)));
+                var group = list.Elements("Year").Elements("Group").Where(g => g.Element("Name").Value.Equals(item.ClassName));
+              
+
```
Comment (svenvanhal) [inline Timetabling/Objects/YearsList.cs]: Newlines

**Comment by svenvanhal** [comment]
Other than that looks good 👍

File: Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs (hunk 1 above)
Comment (karimosman) [reply Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs]: It should be possible to change it, I will need it in an other part.

File: Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
 using Timetabling.DB;
-using Timetabling.Objects;
-using Timetabling.Objects.Constraints.TimeConstraints;
 
-namespace Timetabling.Tests.Objects.Constraints.TimeConstraints.Tests
+namespace Timetabling.Objects.Constraints.TimeConstraints.Tests
 {
     [TestFixture()]
-    internal class ConstraintTeacherSetNotAvailableTimesTest
+    public class ConstraintTeacherSetNotAvailableTimesTest
```
Comment (svenvanhal) [inline Timetabling.Tests/Objects/Constraints/TimeConstraints.Tests/ConstraintTeacherNotAvailableTimesTest.cs]: Is still public atm?
